service/api/prettyprint.go

```diff
@@ -403,14 +416,40 @@ func PrettyExamineMemory(address uintptr, memArea []byte, format byte) string {
 
 	var b strings.Builder
 	w := tabwriter.NewWriter(&b, 0, 0, 3, ' ', 0)
+
 	for i := 0; i < rows; i++ {
 		fmt.Fprintf(w, addrFmt, address)
-		for j := 0; j < cols && i*cols+j < l; j++ {
-			fmt.Fprintf(w, colFormat, memArea[i*cols+j])
+
+		for j := 0; j < cols; j++ {
+			offset := i*(cols*colBytes) + j*colBytes
+			if offset+colBytes <= len(memArea) {
+				data := memArea[offset : offset+colBytes]
+				fmt.Fprintf(w, colFormat, byteArrayToUInt(data))
+			}
 		}
 		fmt.Fprintln(w, "")
 		address += uintptr(cols)
 	}
 	w.Flush()
 	return b.String()
 }
+
+func byteArrayToUInt(arr []byte) interface{} {
```

Comment:
1. the idiomatic way of doing this is returning `(uint, error)`
2. but really the return type should be uint64 or maybe math/big.Int
3. if the former you should clamp the maximum value of size to 8

Reply:
Here use uint64 and limit the maximum of `-size` to 8